src/include/sof/common.h

```diff
@@ -24,11 +24,9 @@
 #include <stddef.h>
 
 /* use same syntax as Linux for simplicity */
-/* Zephyr defines this - remove local copy once Zephyr integration complete */
-#ifdef ARRAY_SIZE
-#undef ARRAY_SIZE
-#endif
+#ifndef __ZEPHYR__ /* Already present and compatible via Zephyr headers */
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+#endif
```

Comment:
wouldn't it be better for all of these to do
```
#ifndef SYMBOL
#define SYMBOL ...
#endif
```
This seems like a more standard and eventually more universal way to me?

Reply:
That's not really what we want though.  We don't want "whatever ARRAY_SIZE happens to have been declared in the local environment", we want one of the two specific implementations that are known to work.